freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/migration/handler/AwsMigrationUtil.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.freeipa.flow.stack.migration.handler;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.services.cloudformation.model.DescribeStackResourcesRequest;
+import com.amazonaws.services.cloudformation.model.DescribeStackResourcesResult;
+import com.amazonaws.services.cloudformation.model.StackResource;
+import com.sequenceiq.cloudbreak.cloud.aws.AwsCloudFormationClient;
+import com.sequenceiq.cloudbreak.cloud.aws.CloudFormationStackUtil;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AwsCredentialView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+
+@Component
+public class AwsMigrationUtil {
+
+    private static final Logger LOGGER = getLogger(AwsMigrationUtil.class);
+
+    @Inject
+    private CloudFormationStackUtil cfStackUtil;
+
+    @Inject
+    private AwsCloudFormationClient awsClient;
+
+    public boolean allInstancesDeletedFromCloudFormation(AuthenticatedContext ac, CloudResource cloudResource) {
+        String regionName = ac.getCloudContext().getLocation().getRegion().value();
+        AwsCredentialView awsCredential = new AwsCredentialView(ac.getCloudCredential());
+        DescribeStackResourcesResult describeStackResourcesResult = awsClient.createCloudFormationClient(awsCredential, regionName)
+                .describeStackResources(new DescribeStackResourcesRequest().withStackName(cloudResource.getName()));
+        List<StackResource> asGroups = describeStackResourcesResult.getStackResources().stream()
+                .filter(it -> "AWS::AutoScaling::AutoScalingGroup".equals(it.getResourceType()))
+                .collect(Collectors.toList());
+        LOGGER.debug("AutoScalingGroup fetched: {}", asGroups);
+        boolean empty = true;
+        int i = 0;
+        while (empty && i < asGroups.size()) {
+            StackResource asGroup = asGroups.get(i);
+            List<String> result = cfStackUtil.getInstanceIds(awsClient.createAutoScalingClient(awsCredential, regionName), asGroup.getPhysicalResourceId());
+            LOGGER.debug("{} autoScalingGroup has {} instance(s): {}", asGroup.getPhysicalResourceId(), result.size(), result);
+            empty = result.isEmpty();
+            i++;
+        }
```

Comment:
this could be refactored into a nice stream like:
```
asgroups.stream().map(fetchresult()).allMatch(result#isEmpty)

Reply:
Is the allMatch guaranteed to exit on the first mismatch?

Reply:
```
Returns whether all elements of this stream match the provided predicate. May not evaluate the predicate on all elements if not necessary for determining the result. If the stream is empty then true is returned and the predicate is not evaluated.
This is a short-circuiting terminal operation.
Params:
predicate – a non-interfering, stateless predicate to apply to elements of this stream
Returns:
true if either all elements of the stream match the provided predicate or the stream is empty, otherwise false
API Note:
This method evaluates the universal quantification of the predicate over the elements of the stream (for all x P(x)). If the stream is empty, the quantification is said to be vacuously satisfied and is always true (regardless of P(x)).

Reply:
I think it is more readable instead of the stream

Reply:
fine, could you please at least use for loop?